### DISCUSSION TITLE
Implementing an array of forms

### DISCUSSION THREAD

**Discussion by alessandrojcm** [react-hook-form/react-hook-form#6175] [Q&A]
Hi there!

At work we need to implement the following feature, but I'm struggling as to what would be the best approach for it. The feature we're working on is a natch payment of products, each row of the table has its own inputs (for price and other data needed to set up the final price of the item), performs its own calculations, has its own payment button, etc. We're using (react-table)[react-table.tanstack.com/] for this, so the code is something along these lines:

```jsx
<table>
    <thead .../>
        <tbody>
             page.map((row) => {
                  // Bunch of not relevant code
                  const { key, ...rowProps } = row.getRowProps();
                  return (
                        <tr {...rowProps}>
                            {row.cells.map((cell) => (
                                // This is the actual render of the inputs, pay buttons, etc
                                <td {...cell.getCellProps()}>{cell.render('Cell')}</Td>
                            ))}
                        </Tr>
                  );
            })
       </tbody>
</table>
```
Now, my idea here is for each row to use an instance of `useForm`; that way each cell of the table can share data using the form context provider, perform validations, use `watch`, have its own submit, etc. I can't do that directly since this is a loop, so I thought about using `useFieldArrays`; but as far is I understood, this is more geared towards forms that have a dynamic number of inputs (and validation is performed against all of them), so this doesn't quite fit my use case since I need each row to be its own separate entity. Any idea on how could I go about this? Thanks!

**Comment by bluebill1049** [comment]
if your forms are fixed, then just use `useForm`, a single instance won't work for you?

**Reply by alessandrojcm** [reply]
Hi, @bluebill1049 thanks for your answer. Sorry, I think I didn't explain myself very well: each row has the same amount of inputs, yes, but the amount of rows is variable and each row has its own state independent of each other row (with its own validation, etc); moreover, if the user clicks pay on one of the rows, then that row should be removed from the table. That's why I think a single `useTable` won't do, since that would hold state for all of the rows as a whole.

**Reply by alessandrojcm** [reply]
Never mind, I managed to work around it refactoring my table a bit; thanks.
